keps/sig-api-machinery/4793-revise-apf-default-configuration/README.md

```diff
@@ -0,0 +1,996 @@
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The APF feature provides flow control to API Server by introducing a limit on
+how much a priority level can consume the concurrency. In addition to that,
+borrowing is also instroduced to better utilize spare concurrency. The limits on
+borrowing are two-sided: a given priority level has a limit on how much it may
+borrow and a limit on how much may be borrowed from it. Worth mentioning, APF
+controller periodically populates current concurrency shares for each priority
+level (i.e. concurrency share that can be borrowed or need to be borrowed) by
+taking past usage into consideration. So in combination of those two aspects, a
+priority level's Nomial Share is only an initial value and the actual share
+might increase or decrease according to actual usage with borrowing. However,
+the owner priority level should have full Nominal Share as its current
+concurrency share if in full utilization.
+
+The current default APF settings are:
+
+Name            | Nominal Shares | Lendable | Proposed Borrowing Limit
+--------------- | -------------: | -------: | -----------------------:
+exempt          | 0              | 50%      | none
+leader-election | 10             | 0%       | none
+node-high       | 40             | 25%      | none
+system          | 30             | 33%      | none
+workload-high   | 40             | 50%      | none
+workload-low    | 100            | 90%      | none
+global-default  | 20             | 50%      | none
+catch-all       | 5              | 0%       | none
+
+This KEP proposes the following settings:
+
+Name            | Nominal Shares | Lendable | Proposed Borrowing Limit
+--------------- | -------------: | -------: | -----------------------:
+exempt          | 0              | 0%       | none
+leader-election | 50             | 80%      | none
+node-high       | 60             | 50%      | none
+system          | 40             | 50%      | none
+workload-high   | 40             | 50%      | none
+workload-low    | 40             | 75%      | none
+global-default  | 10             | 50%      | none
+catch-all       | 5              | 0%       | none
+event           | 5              | 0%       | 100%
+
+Note: The event priority level is specified with Borrowing Limit to prevent it
+from borrowing all spare shares from other priority levels.
+
+In addition to above, a new priority level and flow schema will be created for
+Events:
+
+```
+apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
+kind: FlowSchema
+metadata:
+  name: events
+spec:
+  distinguisherMethod:
+    type: ByUser
+  matchingPrecedence: 450
+  rules:
+  - nonResourceRules:
+    resourceRules:
+    - apiGroups:
+      - events.k8s.io
```

Comment:
Most of components are still using core events.

Reply:
Added, didn't know that. Should we just specify "*" here since the resource is "Event" already?